newm-server/src/main/kotlin/io/newm/server/features/song/model/AgreementMessage.kt

```diff
@@ -0,0 +1,38 @@
+package io.newm.server.features.song.model
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class AgreementMessage(
+    @SerialName("Records")
+    val records: List<Record>?
+) {
+
+    val configurationId: String?
+        get() = records?.firstOrNull()?.s3?.configurationId
```

Comment:
use `by lazy{}` here instead of `get()`. Otherwise each time we get the `configurationId`, it has to do all that null checking again. `by lazy{}` will do the work once and cache the result since the records member should be immutable.

same can be done for `key` below.

Reply:
done